devicetypes/smartthings/zigbee-thermostat.src/zigbee-thermostat.groovy

```diff
@@ -109,92 +116,181 @@ metadata {
+		if (it.cluster == THERMOSTAT_CLUSTER) {
+			if (it.attribute == LOCAL_TEMPERATURE) {
+				log.debug "TEMP"
+				map.name = "temperature"
+				map.value = getTemperature(it.value)
+				map.unit = temperatureScale
+			} else if (it.attribute == COOLING_SETPOINT) {
+				log.debug "COOLING SETPOINT"
+				map.name = "coolingSetpoint"
+				map.value = getTemperature(it.value)
+				map.unit = temperatureScale
+			} else if (it.attribute == HEATING_SETPOINT) {
+				log.debug "HEATING SETPOINT"
+				map.name = "heatingSetpoint"
+				map.value = getTemperature(it.value)
+				map.unit = temperatureScale
+			} else if (it.attribute == THERMOSTAT_MODE || it.attribute == THERMOSTAT_RUNNING_MODE) {
+				log.debug "MODE"
+				map.name = "thermostatMode"
+				map.value = THERMOSTAT_MODE_MAP[it.value]
+				map.data = [supportedThermostatModes: state.supportedThermostatModes]
+			} else if (it.attribute == THERMOSTAT_RUNNING_STATE) {
+				log.debug "RUNNING STATE"
+				def intValue = hexToInt(it.value) as int
+				/**
+				 * Zigbee Cluster Library spec 6.3.2.2.3.7
+				 * Bit	Description
+				 *  0	Heat State
+				 *  1	Cool State
+				 *  2	Fan State
+				 *  3	Heat 2nd Stage State
+				 *  4	Cool 2nd Stage State
+				 *  5	Fan 2nd Stage State
+				 *  6	Fan 3rd Stage Stage
+				 **/
+				map.name = "thermostatOperatingState"
+				if (intValue & 0x01) {
+					map.value = "heating"
+				} else if (intValue & 0x02) {
+					map.value = "cooling"
+				} else if (intValue & 0x04) {
+					map.value = "fan only"
+				} else {
+					map.value = "idle"
+				}
+			} else if (it.attribute == CONTROL_SEQUENCE_OF_OPERATION) {
+				log.debug "CONTROL SEQUENCE OF OPERATION"
+				state.supportedThermostatModes = CONTROL_SEQUENCE_OF_OPERATION_MAP[it.value]
+				map.name = "supportedThermostatModes"
+				map.value = JsonOutput.toJson(CONTROL_SEQUENCE_OF_OPERATION_MAP[it.value])
+			}
+			// Thermostat System Config is an optional attribute, but is supported by the LUX KONOz and is more informative.
+			else if (it.attribute == THERMOSTAT_SYSTEM_CONFIG) {
+				log.debug "THERMOSTAT SYSTEM CONFIG"
+				def intValue = hexToInt(it.value) as int
+				/**
+				 *
+				 * Table 6-12. HVAC System Type Configuration Values
+				 * Bit Number	Description
+				 * 	0 – 1		Cooling System Stage
+				 * 					00 – Cool Stage 1
+				 * 					01 – Cool Stage 2
+				 * 					10 – Cool Stage 3
+				 * 					11 – Reserved
+				 * 	2 – 3		Heating System Stage
+				 * 					00 – Heat Stage 1
+				 * 					01 – Heat Stage 2
+				 * 					10 – Heat Stage 3
+				 * 					11 – Reserved
+				 * 	4			Heating System Type
+				 * 					0 – Conventional
+				 * 					1 – Heat Pump
+				 * 	5			Heating Fuel Source
+				 * 					0 – Electric / B
+				 * 					1 – Gas / O
+				 */
+				def cooling = 	   intValue & 0b00000011
+				def heating = 	  (intValue & 0b00001100) >>> 2
+				def heatingType = (intValue & 0b00010000) >>> 4
+				def supportedModes = ["off"]
+
+				if (cooling != 0x03) {
```

Comment:
It would be nice to make some of these numbers constants with a descriptive name. Then it will be more clear what is being checked.

Reply:
I'll look into doing another PR that does some cleanup. Thanks!